src/hedera/hederaEngine.js

```diff
@@ -0,0 +1,404 @@
+
+        if (!this.walletLocalData.otherData.activationPaymentUrl) {
+          throw new Error({
+            message: 'ErrorNoActivationPending'
+          })
+        }
+
+        const options = {
+          method: 'PUT',
+          headers: {
+            Accept: 'application/octet-stream',
+            'Content-Type': 'application/octet-stream'
+          },
+          body: hexToBuf(txn.signedTx)
+        }
+        const response = await (
+          await this.io.fetch(
+            this.walletLocalData.otherData.activationPaymentUrl,
+            options
+          )
+        ).json()
+
+        if (response.status === 'success') {
+          this.walletLocalData.otherData.activationStatusUrl =
+            response.status_url
+          this.addToLoop('checkAccountCreationStatus', 5000)
+        } else {
+          this.log('ErrorInvalidActivationPayment', response.error)
+          throw new Error('ErrorInvalidActivationPayment')
+        }
+      }
+    }
+  }
+
+  async checkAccountCreationStatus(): Promise<void> {
+    const statusUrl = this.walletLocalData.otherData.activationStatusUrl
+
+    if (!statusUrl || this.walletLocalData.otherData.hederaAccount) {
+      return
+    }
+
+    this.log('checkAccountCreationStatus', statusUrl)
+
+    let response
+
+    try {
+      response = await (await this.io.fetch(statusUrl)).json()
+    } catch (e) {
+      this.log('error checking Hedera account creation status', e)
+      return
+    }
+
+    if (response.status === 'error') {
+      delete this.timers.checkAccountCreationStatus
+      throw new Error({
+        message: 'ErrorAccountActivation',
+        reason: response.message
+      })
+    }
+
+    if (response.status === 'success') {
+      this.walletLocalData.otherData.hederaAccount = response.account_id
+      this.walletLocalDataDirty = true
+      this.accountId = new hedera.AccountId(response.account_id)
+      await this.saveWalletLoop()
+
+      delete this.timers.checkAccountCreationStatus
+    }
+  }
+
+  // ****************************************************************************
+  // Public methods
+  // ****************************************************************************
+
+  async loadEngine(
+    plugin: EdgeCurrencyTools,
+    walletInfo: EdgeWalletInfo,
+    opts: EdgeCurrencyEngineOptions
+  ): Promise<void> {
+    await super.loadEngine(plugin, walletInfo, opts)
+
+    if (!this.walletLocalData.otherData.hederaAccount) {
+      const accountId = await this.getAccountId()
+
+      if (accountId) {
+        this.accountId = accountId
+        this.walletLocalData.otherData.hederaAccount = accountId.toString()
+        this.walletLocalDataDirty = true
+      } else {
+        await this.checkAccountCreationStatus()
+      }
+    } else {
+      this.accountId = new hedera.AccountId(
+        this.walletLocalData.otherData.hederaAccount
+      )
+    }
+  }
+
+  async startEngine() {
+    console.log('hederaEngine/startEngine', this)
```

Comment:
Do not console.log whole objects. This gets messy and could be a security issue.

Reply:
This was for debugging during development, I'll be removing most of these log calls.